containers/cpp-mariadb/.devcontainer/install-mariadb.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+
+. /etc/os-release
+
+MARIADB_REPO_SETUP=mariadb_repo_setup
+TMP_DIR=$(mktemp -d -t maria-XXXXXXXXXX)
+MARIADB_CONNECTOR=""
+
+function cleanup {
+    cd /
+    rm -rf ${TMP_DIR}/${MARIADB_REPO_SETUP}
+    rm -rf ${TMP_DIR}/${MARIADB_CONNECTOR}*
+}
+trap cleanup EXIT
```

Comment:
Since you are executing this cleanup function on exit, it may be worth:
- Returning the exit code
- adding a `+e` before removing the tmp dir so this doesn't fail if there is an issue

For example:

```sh
cleanup( ) {
    EXIT_CODE=$?
    set +e
    if [[ -n "${TMP_DIR}" ]]; then
         # cd /
         # remove all the tmp dir things
    fi
    exit $EXIT_CODE
}
trap cleanup EXIT
```